backend/data_preparation/extractor/fire_extractor.py

```diff
@@ -0,0 +1,203 @@
+        else:
+            # after 2016
+            # For some records after 2016, record schema for firename is: FIRENAME, for agency is: AGENCY, for
+            # datetime is: PERDATTIME
+            data["firename"] = record.as_dict().get("FIRE_NAME",  record.as_dict().get('fireName',
+                                                                                record.as_dict().get("FIRENAME")))
+            data["agency"] = record.as_dict().get("AGENCY", record.as_dict().get("agency")) \
+                if record.as_dict().get("AGENCY", record.as_dict().get("agency")) != "" else "Unknown"
+            data["datetime"] = FireExtractor._get_datetime_after_2016(record)
+            # after 2016, record schema for area is: GISACRES
+            # For some records after 2016, record schema for area is: gisAcres
+            data["area"] = float(record.as_dict().get("GISACRES", record.as_dict().get("gisAcres")))
+        geom = FireExtractor._extract_full_geom(shapefile_reader)
+        # basic geom is the full geom of this shapefile record
+        # see function extract_full_geom below
+        data["geopolygon_full"] = str(geom)
+        # geopolygon_full is the full geom of this shapefile record
+        data["geopolygon_large"] = str(FireExtractor._simplify_multipolygon(geom,1.e-04))
+        # geopolygon_large is the simplified full geom of this shapefile record, threshold is 1.e-04
+        data["geopolygon_medium"] = str(FireExtractor._simplify_multipolygon(geom,1.e-03))
+        # geopolygon_medium is the simplified full geom of this shapefile record, threshold is 1.e-03
+        data["geopolygon_small"] = str(FireExtractor._simplify_multipolygon(geom,1.e-02))
+        # geopolygon_small is the simplified full geom of this shapefile record, threshold is 1.e-02
+        data["is_sequential"] = is_sequential
+        # is_sequential is passed as a parameter. If in the temp folder, there are more than one set of records, then
+        # is_sequential is True
+        data["fire_id"] = fire_id
+        # fire_id is passed as a parameter
+        data["state"] = state
+        # state is passed as a parameter
+        logger.info(f"Finished extracting fire record: {path}")
+        return data
+
+    @staticmethod
+    def _extract_full_geom(shp: shapefile.Reader) -> MultiPolygon:
+        """
+        Extracts a full geom from a shp reader
+        :param shp: shapefile.Reader
+        :return: Multipolygon
+        """
+        return shape(shp.shapeRecord(0).shape.__geo_interface__)
+
+    @staticmethod
+    def _simplify_multipolygon(multipolygon: MultiPolygon, threshold: float) -> MultiPolygon:
+        """
+        Simplifies all components of a multipolygon
+        :param multipolygon:shapely.geometry.Multipolygon or shapely.geometry.polygon.Polygon
+        :param threshold:float, the threshold of simplification
+        :return:shapely.geometry.Multipolygon
+        """
+        if multipolygon is MultiPolygon:
```

Comment:
this is logically wrong. 

please use `if isinstance(multipolygon , MultiPolygon)` instead.

`is` in python is checking if objects' id are matching.